File: core/src/main/java/com/vzome/core/edits/ValidateSelection.java

package com.vzome.core.edits;

import com.vzome.core.commands.Command.Failure;
import com.vzome.core.editor.api.ChangeSelection;
import com.vzome.core.editor.api.EditorModel;


public class ValidateSelection extends ChangeSelection
{
    @Override
    public void perform() throws Failure
    {
        if ( mSelection .size() == 0 )
            throw new Failure( "selection is empty" );
    }

    public ValidateSelection( EditorModel editor )
    {
        super( editor .getSelection() ); 
    }

    @Override
    protected String getXmlElementName()
    {
        return "ValidateSelection";
    }

}

File: online/src/worker/legacy/ts/com/vzome/core/edits/ValidateSelection.ts
import { Command } from "../commands/Command.js";
import { ChangeSelection } from "../editor/api/ChangeSelection.js";
import { EditorModel } from "../editor/api/EditorModel.js";

export class ValidateSelection extends ChangeSelection {
    /**
     * 
     */
    public perform() {
        if (this.mSelection.size() === 0)throw new Command.Failure("selection is empty");
    }

    public constructor(editor: EditorModel) {
        super(editor.getSelection());
    }

    /**
     * 
     * @return {string}
     */
    getXmlElementName(): string {
        return "ValidateSelection";
    }
}
ValidateSelection["__class"] = "com.vzome.core.edits.ValidateSelection";
